File: src/types/session.ts
import type { PhotoEntry, PhotoState } from './image';

export interface SessionFile {
  id: string;
  folderPath: string;
  recursive: boolean;
  createdAt: number;
  updatedAt: number;
  currentIndex: number;
  images: PhotoEntry[];
  states: Record<string, PhotoState>;
}

export interface SessionStats {
  total: number;
  selected: number;
  rejected: number;
  favorites: number;
  skipped: number;
  pending: number;
  completionPercent: number;
  avgSecondsPerPhoto: number | null;
  estimatedSecondsRemaining: number | null;
}

export interface ScanResult {
  images: PhotoEntry[];
  total: number;
  elapsedMs: number;
}

File: src-tauri/src/models/session.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

use super::photo::{PhotoEntry, PhotoState};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct SessionFile {
    pub id: String,
    pub folder_path: String,
    pub recursive: bool,
    pub created_at: i64,
    pub updated_at: i64,
    pub current_index: i64,
    pub images: Vec<PhotoEntry>,
    pub states: HashMap<String, PhotoState>,
}
